application/notifications/sendgrid.go

```diff
@@ -42,29 +43,23 @@ func (e *SendGridService) Send(msg Message) error {
 		return errors.New("SendGrid API key is required")
 	}
 
-	template, ok := sendGridTemplates[msg.Template]
-	if !ok {
-		return fmt.Errorf("invalid message template name: %s", msg.Template)
-	}
-
-	p := mail.NewPersonalization()
-	p.AddTos(mail.NewEmail(msg.ToName, msg.ToEmail))
-	for key, val := range msg.Data {
-		p.SetDynamicTemplateData(key, val)
-	}
+	subject := "New Message on " + domain.Env.AppName
+	from := mail.NewEmail(msg.FromName, msg.FromEmail)
+	to := mail.NewEmail(msg.ToName, msg.ToEmail)
 
-	message := mail.NewV3Mail()
-	message.SetTemplateID(template.id)
-	message.SetFrom(mail.NewEmail(msg.FromName, msg.FromEmail))
-	message.AddPersonalizations(p)
+	msg.Data["uiURL"] = domain.Env.UIURL
+	msg.Data["appName"] = domain.Env.AppName
 
-	domain.Logger.Printf("email data: %+v\n", message.Personalizations[0].DynamicTemplateData)
+	bodyBuf := &bytes.Buffer{}
+	if err := r.HTML(msg.Template).Render(bodyBuf, msg.Data); err != nil {
+		return errors.New("error rendering message body - " + err.Error())
+	}
+	body := bodyBuf.String()
 
-	request := sendgrid.GetRequest(apiKey, SendGridEndpointMailSend, SendGridAPIUrl)
-	request.Method = "POST"
-	request.Body = mail.GetRequestBody(message)
+	m := mail.NewSingleEmail(from, subject, to, body, body)
```

Comment:
Are you stripping HTML tags for the plain-text version? I have some code in a stash that does this if you're interested.